### PR TITLE
fix(release): add branch env var, protect against release loops

### PR DESCRIPTION
**Pull request by hayemaxi**
## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: buildspec/release/00clonerepo.yml [hunk 1]
```diff
@@ -13,12 +13,23 @@ phases:
...
-            - cd aws-toolkit-vscode && git checkout master
+            # checkout the target branch as we want to commit to it later to update versions
+            - cd aws-toolkit-vscode && git checkout ${TARGET_BRANCH}
+
+            # Since we don't have a manual gate on the amazon q release pipeline, we must make sure we aren't
+            # releasing off of a commit from another release, i.e. infinite loop.
+            # This will be removed prior to standalone extension release.
+            - |
+                AUTHOR_AUTOMATION="aws-toolkit-automation"
+                if [[ $(git log -1 --pretty=%an) = $AUTHOR_AUTOMATION && $(git log -1 --pretty=%B) != "Merge master into feature/standalone" ]]; then
```
Comment (leigaol) [inline buildspec/release/00clonerepo.yml:29]: "Merge master into feature/standalone" 

How about commit in the format of "Release 2.18.0"

Comment (leigaol) [inline buildspec/release/00clonerepo.yml:29]: When there is a merge conflict, the merge message may not be "Merge master into feature/standalone". 

Is there a more robust way to detect if the last commit is from pipeline?

Comment (justinmk3) [reply buildspec/release/00clonerepo.yml:29]: could use `grep` to check for `[mM]erge`. Following is untested:
```suggestion
                if [[ $(git log -1 --pretty=%an) = $AUTHOR_AUTOMATION && git log -1 --pretty=%B | >/dev/null grep '[mM]erge' ]]; then
```

Comment (hayemaxi) [reply buildspec/release/00clonerepo.yml]: I just removed this block. Releasing like this is just too risky, will have to deal with race conflicts. Will update the release pipeline to have an alternative method.

**Review by justinmk3 — APPROVED** [review]

**Pull request merged**